src/NHibernate/Loader/Loader.cs

```diff
@@ -982,15 +984,8 @@ private void CheckVersion(int i, IEntityPersister persister, object id, object e
 				object obj = null;
 				EntityKey key = keys[i];
 
-				if (key == null)
-				{
-					// do nothing
-					/* TODO NH-1001 : if (persisters[i]...EntityType) is an OneToMany or a ManyToOne and
-					 * the keys.length > 1 and the relation IsIgnoreNotFound probably we are in presence of
-					 * an load with "outer join" the relation can be considerer loaded even if the key is null (mean not found)
-					*/
-				}
-				else
+				// if key is null do nothing
```

Comment:
Maybe instead of commenting the obvious we should clarify where null key is handled. Something like:
```suggestion
				// null keys are handled in RegisterNonExists
```